### PR TITLE
Backport 2.28 - Add tests for an opaque import in the driver wrappers 

### PR DESCRIPTION
**Pull request by AndrzejKurek**
Backport of https://github.com/ARMmbed/mbedtls/pull/5282.

### CODE REVIEW

File: tests/src/drivers/test_driver_key_management.c [hunk 1]
```diff
@@ -229,14 +232,30 @@ psa_status_t mbedtls_test_opaque_export_key(
     const uint8_t *key, size_t key_length,
     uint8_t *data, size_t data_size, size_t *data_length )
 {
+    psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
```
Comment (ronald-cron-arm) [inline tests/src/drivers/test_driver_key_management.c:235]: We should not change `mbedtls_test_opaque_export_key`. `mbedtls_test_opaque_import_key` is not implemented in 2.28 thus nothing to do in fact in the opaque case.

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]

File: tests/src/drivers/test_driver_key_management.c (hunk 1 above)
Comment (AndrzejKurek) [reply tests/src/drivers/test_driver_key_management.c:235]: Automatic conflict resolver error. My bad for not checking that.

File: tests/src/drivers/test_driver_key_management.c [hunk 2]
```diff
@@ -229,15 +232,15 @@ psa_status_t mbedtls_test_opaque_export_key(
     const uint8_t *key, size_t key_length,
     uint8_t *data, size_t data_size, size_t *data_length )
 {
+    /* Assume this is a builtin key based on the key material length. */
```
Comment (ronald-cron-arm) [inline tests/src/drivers/test_driver_key_management.c:235]: This was not changed in 5282 thus no point in changing it in this backport I'd say.

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]

File: tests/src/drivers/test_driver_key_management.c (hunk 2 above)
Comment (AndrzejKurek) [reply tests/src/drivers/test_driver_key_management.c:235]: The structure of the `mbedtls_test_opaque_export_key` is different in the `development`, and the variable is declared there in a smaller scope, so I would rather consider this fix applicable to 2.28 only:

```
    if( key_length == sizeof( psa_drv_slot_number_t ) )
    {
        /* Assume this is a builtin key based on the key material length. */
        psa_drv_slot_number_t slot_number = *( ( psa_drv_slot_number_t* ) key );

        switch( slot_number )
        {
			...
        }
    }
    else
    {
        ...
    }
    return( PSA_ERROR_NOT_SUPPORTED );
```
If you still want me to remove it, I'll do it, but I think that it's a good improvement.

Comment (ronald-cron-arm) [reply tests/src/drivers/test_driver_key_management.c:235]: ok, this fits well in commit "Test driver: keep variable declarations first " thus ok to keep it.

**Review by ronald-cron-arm — APPROVED** [review]
LGTM

**Comment by ronald-cron-arm** [comment]
@mprse as you reviewed #5282, could you review its backport to 2.28 ? The backport is trivial thus I prefer to have two reviewers.

**Review by mprse — APPROVED** [review]
LGTM

**Pull request merged**